services/openvpn/connection_factory.go

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package openvpn
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/mysteriumnetwork/go-openvpn/openvpn"
+	"github.com/mysteriumnetwork/go-openvpn/openvpn/management"
+	"github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/client/auth"
+	openvpn_bytescount "github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/client/bytescount"
+	"github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/state"
+	"github.com/mysteriumnetwork/node/client/stats"
+	"github.com/mysteriumnetwork/node/core/connection"
+	"github.com/mysteriumnetwork/node/core/location"
+	"github.com/mysteriumnetwork/node/identity"
+	"github.com/mysteriumnetwork/node/server"
+	"github.com/mysteriumnetwork/node/services/openvpn/middlewares/client/bytescount"
+	openvpn_session "github.com/mysteriumnetwork/node/services/openvpn/session"
+	"github.com/mysteriumnetwork/node/session"
+)
+
+// OpenvpnProcessBasedConnectionFactory represents a factory for creating process-based openvpn connections
+type OpenvpnProcessBasedConnectionFactory struct {
+	mysteriumAPIClient    server.Client
+	openvpnBinary         string
+	configDirectory       string
+	runtimeDirectory      string
+	originalLocationCache location.Cache
+	signerFactory         identity.SignerFactory
+	statsKeeper           stats.SessionStatsKeeper
+}
+
+// NewOpenvpnProcessBasedConnectionFactory creates a new OpenvpnProcessBasedConnectionFactory
+func NewOpenvpnProcessBasedConnectionFactory(
```

Comment:
we try to get rid of New..Factory() methods, we should proceed this this at this point.